pages/expense.js

```diff
@@ -260,9 +254,17 @@ class ExpensePage extends React.Component {
     const parentCollectiveSlugRoute = parentCollectiveSlug ? `${parentCollectiveSlug}/` : '';
     const collectiveType = parentCollectiveSlug ? getCollectiveTypeForUrl(data?.account) : undefined;
     const collectiveTypeRoute = collectiveType ? `${collectiveType}/` : '';
-    this.props.router.push(
```

Comment:
@kewitz I was not sure how to test this `handleExpenseVerification` path, do you have any tips?

Reply:
@Betree there's a test suite in `27-expense.test.js` file called `submit on behalf` that will catch this behavior.
To test it manually, you can:
1. Create a new expense;
2. Invite a new user;
3. Open the invite link logged in your API console in an incognito window;
4. Sign up and submit the expense;
5. Open the login link logged in your API console;
    - You should be logged in and redirected to the expense view and **the expense should be marked as pending**.

Reply:
Thanks!